storage/remote/write.go

```diff
@@ -94,25 +92,10 @@ func (rws *WriteStorage) ApplyConfig(conf *config.Config) error {
 	rws.mtx.Lock()
 	defer rws.mtx.Unlock()
 
-	configHash, err := toHash(conf.RemoteWriteConfigs)
```

Comment:
is this not useful to keep to just short circuit the rest of the function if there hasn't been any config changes?

Reply:
Same reason as below due to the redaction of secrets keeping it as a hash doesn't tell us if any secrets changed. We could possibly keep the entire value in memory and do a `DeepEqual` on it, but the optimization seemed relatively minor, and I was running into some DeepEqual issues comparing clients.